### ISSUE TITLE
Depends.py enhancement: removal of the complementary domain

### ISSUE THREAD

**Issue by AndrzejKurek** [Mbed-TLS/mbedtls#6438]
After https://github.com/Mbed-TLS/mbedtls/pull/6385, there are two domains that use the complementary domain:
- hashes (!MBEDTLS_MD5_C, !MBEDTLS_RIPEMD160_C, !MBEDTLS_SHA1_C, !MBEDTLS_SHA224_C, !MBEDTLS_SHA384_C, MBEDTLS_SHA224_C, MBEDTLS_SHA384_C);
- pkalgs (!MBEDTLS_ECDSA_C, !MBEDTLS_ECP_C, !MBEDTLS_PKCS1_V21, !MBEDTLS_PKCS1_V15, !MBEDTLS_RSA_C, !MBEDTLS_X509_RSASSA_PSS_SUPPORT);
The task is to:
- switch to ExclusiveDomain for pkalgs;
- determine whether it's necessary to test hashes using the ComplementaryDomain;
- fix all issues that arise.

**Comment by mpg** [comment]
Note: while at it we also need to make sure we test a build with `MBEDTLS_ECDSA_C` but without `MBEDTLS_ECDSA_DETERMINSTIC`. This was previously tracked as https://github.com/Mbed-TLS/mbedtls/issues/2203 which I'm closing in favour of this one.

Generally speaking, ExclusiveDomain for PK algs is a a bit tricky as in some case we won't be really exclusive:
- `MBEDTLS_ECDSA_DETERMINSTIC` depends on `MBEDTLS_ECDSA_C` - what we want is test the later with and without the former
- `MBEDTLS_X509_RSASSA_PSS_SUPPORT` depends on `MBEDTLS_PKCS1_V21` and again we want to test the later with and without the former
- similarly again for `MBEDTSL_ECDSA_C` that depends on `MBEDTLS_ECP_C`

So, "only ECP" would mean just that, but "only ECDSA" would mean ECDSA+ECP, and "only ECDSA derministic" would mean "deterministic+ECDSA+ECP".

**Comment by gilles-peskine-arm** [comment]
I thought we were moving away from the domain concept and our only dimension is “just one thing”? Just one thing, in complex cases, has to include the dependencies of the thing. So e.g. deterministic ECDSA = `MBEDTLS_ECDSA_DETERMINSTIC` + everything it needs (ECDSA, ECP, HMAC_DRBG) + all the stuff that's not related to pk.

**Comment by mpg** [comment]
Yes, that's basically what I was trying to say. What did you think I meant? (So that perhaps in the future I can avoid misunderstanding.)
